hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/token/AbstractTokenAirdrop.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2019-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.common.domain.token;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.Range;
+import com.hedera.mirror.common.domain.History;
+import com.hedera.mirror.common.domain.UpsertColumn;
+import com.hedera.mirror.common.domain.Upsertable;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.MappedSuperclass;
+import java.io.Serial;
+import java.io.Serializable;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+
+@Data
+@IdClass(AbstractTokenAirdrop.Id.class)
+@MappedSuperclass
+@NoArgsConstructor
+@SuperBuilder(toBuilder = true)
+@Upsertable(history = true)
+public class AbstractTokenAirdrop implements History {
+
+    @UpsertColumn(
+            coalesce =
+                    """
+                    case when coalesce(state, e_state) = ''PENDING'' then coalesce(e_{0}, 0) + coalesce({0}, 0)
+                            else coalesce(e_{0}, {0}, 0)
+                        end""")
+    private long amount;
```

Comment:
```suggestion
    private Long amount;
```

so for NFT airdrop, the value is set to `NULL`

Reply:
Got it I have changed amount to Long.